extensions-core/druid-basic-security/src/main/java/io/druid/security/basic/CommonCacheNotifier.java

```diff
@@ -0,0 +1,238 @@
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.security.basic;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import com.metamx.emitter.EmittingLogger;
+import com.metamx.http.client.HttpClient;
+import com.metamx.http.client.Request;
+import com.metamx.http.client.response.ClientResponse;
+import com.metamx.http.client.response.HttpResponseHandler;
+import com.metamx.http.client.response.StatusResponseHolder;
+import io.druid.discovery.DiscoveryDruidNode;
+import io.druid.discovery.DruidNodeDiscovery;
+import io.druid.discovery.DruidNodeDiscoveryProvider;
+import io.druid.java.util.common.StringUtils;
+import io.druid.java.util.common.concurrent.Execs;
+import io.druid.server.DruidNode;
+import org.jboss.netty.handler.codec.http.HttpChunk;
+import org.jboss.netty.handler.codec.http.HttpMethod;
+import org.jboss.netty.handler.codec.http.HttpResponse;
+import org.joda.time.Duration;
+
+import javax.ws.rs.core.MediaType;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+public class CommonCacheNotifier
+{
+  private static final List<String> NODE_TYPES = Arrays.asList(
+      DruidNodeDiscoveryProvider.NODE_TYPE_BROKER,
+      DruidNodeDiscoveryProvider.NODE_TYPE_OVERLORD,
+      DruidNodeDiscoveryProvider.NODE_TYPE_HISTORICAL,
+      DruidNodeDiscoveryProvider.NODE_TYPE_PEON,
+      DruidNodeDiscoveryProvider.NODE_TYPE_ROUTER,
+      DruidNodeDiscoveryProvider.NODE_TYPE_MM
+  );
+
+  private final DruidNodeDiscoveryProvider discoveryProvider;
+  private final HttpClient httpClient;
+  private final Set<String> itemsToUpdate;
+  private final Map<String, byte[]> serializedMaps;
+  private final Map<String, BasicAuthDBConfig> itemConfigMap;
+  private final String baseUrl;
+  private final EmittingLogger logger;
+  private final String threadName;
+
+  private Thread notifierThread;
+
+  public CommonCacheNotifier(
+      Map<String, BasicAuthDBConfig> itemConfigMap,
+      DruidNodeDiscoveryProvider discoveryProvider,
+      HttpClient httpClient,
+      String baseUrl,
+      String threadName,
+      EmittingLogger logger
```

Comment:
hm, I feel like it'd be more useful to use this logger passed in from the authenticator/authorization classes, so that the logs appear differently based on the class that's using this common notifier

Reply:
Ah it's needed. But I'm still concerned with that it's difficult to find where the logging code is from the log if the logging class is different from the actual class. What do you think about passing a caller name in constructor and adding it to all logs?